tasks/agent.py

```diff
@@ -267,7 +268,14 @@ def run(
     if not skip_build:
         build(ctx, rebuild, race, build_include, build_exclude, flavor)
 
-    ctx.run(os.path.join(BIN_PATH, bin_name("agent")))
+    cmd = os.path.join(BIN_PATH, bin_name("agent"))
+    cmd += ' run -c '
+    if config_path is None:
```

Comment:
We might just want to check if `config_path` is set to prevent using an empty string:

```suggestion
    if config_path:
```